src/modules/job-manager/event.c

```diff
@@ -405,6 +405,7 @@ int event_job_action (struct event *event, struct job *job)
                                 (uintmax_t) job->id);
             }
             (void) jobtap_call (ctx->jobtap, job, "job.destroy", NULL);
+            job_aux_destroy (job);
```

Comment:
Since the aux items for plugins are of a known form, we could also potentially search the aux_item list for all entries that look like plugin entries and just destroy _those_. However, since jobtap plugins are the only user of `job->aux` at this point, that might be a bit of a waste until necessary.

Reply:
It seemed to me that there's a reasonable chance we would want to clean up any future  non-plugin use case too, so I thought maybe we should leave it like this for now.

Reply:
This makes sense. I wasn't sure if a user within the job manager (e.g. outside of any plugin) would expect to be able to access aux data even for inactive jobs.

Reply:
Well for now we can leave it this way with the comment in the header and later we could change it if need be.